Add collected-notes extension

txemaleon opened the pull request:
## Description

Collected Notes is an online notes system which lets you publish as a blog. It offers an API which this extension connects to and let's you manage your notes, edit, set visibility, or delete them entirely.

## Screencast

All the functionality is covered on the metadata screenshots.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- If using wrong credentials it never stops loading, it should stop loading and let the user know if possible.

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

txemaleon commented:
Good morning.

I've added validation to every command, now it checks wether you're properly logged in.

I've also pulled your contributions properly, I feel the subtitle in one of the commands does not fit well since it's the same as the title.

![Raycast 2024-01-15 at 08 26 21](https://github.com/raycast/extensions/assets/5676055/4cc4b508-c7b8-43a2-991a-65e712e0d6f9)

Hope it fits the store.

pernielsentikaer commented:
You're right about the subtitle, you could also remove it from the second again and call it `Create Collected Note` instead 🙂

It would be okay if it just failed when you tried to create the note, right not I get almost 2 seconds before the validation for that command show up 🙂

Awesome work 🔥

pernielsentikaer reviewed (changes requested):
As stated above 🙂

txemaleon commented:
I've renamed both commands and removed the subtitles -which still shows the command name-.

I've also removed the validation in the create-note command but instead made it to show a toast if there is any problem.

Validation also must complete before creating the note.

pernielsentikaer commented on extensions/collected-notes/src/layouts/InvalidCredentials.tsx:
```diff
@@ -0,0 +1,16 @@
+			markdown={message ?? markdown}
+			actions={
+				<ActionPanel>
+					<Action title="Open Extension Preferences" onAction={openExtensionPreferences} />
```
```suggestion
					<Action title="Open Extension Preferences" icon={Icon.Gear} onAction={openExtensionPreferences} />
```

pernielsentikaer commented on extensions/collected-notes/src/layouts/InvalidCredentials.tsx:
```diff
@@ -0,0 +1,16 @@
+import { Action, ActionPanel, Detail, openExtensionPreferences } from "@raycast/api";
```
```suggestion
import { Action, ActionPanel, Detail, Icon, openExtensionPreferences } from "@raycast/api";
```

pernielsentikaer commented on extensions/collected-notes/.prettierrc:
```diff
@@ -0,0 +1,5 @@
+{
+	"printWidth": 120,
+	"singleQuote": false,
+	"useTabs": true
```
```suggestion
```
Could you remove this so we have the same settings for all extensions, it might require a fix-lint to be ran from package.json 🙂

pernielsentikaer reviewed (changes requested):
Thanks for the corrections 🔥

While being in development mode it might show the name in subtitle, when it's build it shoildn't

I left a few comments for you 🙂

- Could you make sure that actions panel items all have icons 🙂

- Check my suggestions

txemaleon commented:
I've merged your changes and also made a little adjustment to the `Create Note` Action to have an icon an a proper title whenever you want to send the note.

I've also run a `fix-lint` with no tabs.

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥

txemaleon commented:
Wondeful! Thanks!

alex-cory commented:
Would be super cool if you could sync up a Notion db for this
